### PR TITLE
rust/zh Add chinese translation for rust

### PR DESCRIPTION
**Pull request by Garnel**
Hi, it is a Chinese translation for rust

### CODE REVIEW

File: zh-cn/rust-cn.html.markdown [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    let x: &str = "hello world!";
+
+    // 输出
+    println!("{} {}", f, x); // 1.3 hello world
+
+    // 一个 `String` – 在堆上分配空间的字符串
+    let s: String = "hello world".to_string();
+
+    // 字符串分片(slice) - 另一个字符串的不可变视图
+    // 基本上就是指向一个字符串的不可变指针，它不包含字符串里任何类容，只是一个指向某个东西的指针
```
Comment (geoffliu) [inline zh-cn/rust-cn.html.markdown]: 类容?

File: zh-cn/rust-cn.html.markdown [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        fn frobnicate(self) -> Option<T> {
+            Some(self.bar)
+        }
+    }
+
+    let another_foo = Foo { bar: 1 };
+    println!("{:?}", another_foo.frobnicate()); // Some(1)
+
+    ///////////////////////////////////
+    // 3. 模板匹配 (Pattern matching) //
```
Comment (geoffliu) [inline zh-cn/rust-cn.html.markdown]: 模式匹配 seems to be the more common translation.

File: zh-cn/rust-cn.html.markdown [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        FooBar { x: n, y: OptionalI32::AnI32(m) } if n == m =>
+            println!("The numbers are the same"),
+        FooBar { x: n, y: OptionalI32::AnI32(m) } =>
+            println!("Different numbers: {} {}", n, m),
+        FooBar { x: _, y: OptionalI32::Nothing } =>
+            println!("The second number is Nothing!"),
+    }
+
+    ///////////////////////////////
+    // 4. 条件控制 (Control flow) //
```
Comment (geoffliu) [inline zh-cn/rust-cn.html.markdown]: 流程控制

File: zh-cn/rust-cn.html.markdown [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    loop {
+        println!("Hello!");
+    }
+
+    ////////////////////////////////////////////////
+    // 5. 内存安全和指针 (Memory safety & pointers) //
+    ////////////////////////////////////////////////
+
+    // 独占指针 (Owned pointer) - 同一时刻只能有一个对象能“拥有”这个指针
+    // 意味着 `Box` 离开他的作用域后，会被安全的释放
```
Comment (geoffliu) [inline zh-cn/rust-cn.html.markdown]: 安全地释放

File: zh-cn/rust-cn.html.markdown [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+
+    println!("{}", *ref_var2); // 6
+    // var2 = 2; // 编译报错，因为 `var2` 被借用了
+}
+```
+
+## 更深入的资料
+
+Rust 还有很多很多其他类容 - 这只是 Rust 最基础的东西，帮助你了解 Rust 里面最重要的东西。
+如果想深入学习 Rust，可以去阅读
```
Comment (geoffliu) [inline zh-cn/rust-cn.html.markdown]: Nit: 可以读

File: zh-cn/rust-cn.html.markdown [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+    *ref_var2 += 2;
+
+    println!("{}", *ref_var2); // 6
+    // var2 = 2; // 编译报错，因为 `var2` 被借用了
+}
+```
+
+## 更深入的资料
+
+Rust 还有很多很多其他类容 - 这只是 Rust 最基础的东西，帮助你了解 Rust 里面最重要的东西。
```
Comment (geoffliu) [inline zh-cn/rust-cn.html.markdown]: You're repeating “东西” here. Maybe change the first to say “最基础的功能”

**Comment by geoffliu** [comment]
Hey @Garnel, thanks for the hard work. This looks pretty great, just a few small problems I found. We can get this merged as soon as we get those fixed.

**Comment by Garnel** [comment]
@geoffliu thanks for the review, I will fix those problems today

**Comment by geoffliu** [comment]
Nice, thanks!

**Pull request merged**